Update kcidb documentation

nuclearcat opened the pull request:
With introduction of kcidb-rest we need to update documentation, and even before that it need to be updated about CloudSQL role, and fact that BigQuery is mostly deprecated.

nuclearcat committed:
Update kcidb documentation

With introduction of kcidb-rest we need to update documentation,
and even before that it need to be updated about CloudSQL role,
and fact that BigQuery is mostly deprecated.

Signed-off-by: Denys Fedoryshchenko <denys.f@collabora.com>

padovan reviewed (approved):
It looks good to me! It describes the transition well.

nuclearcat committed:
(doc/submitter_guide.md) Remove deprecated field, add info about REST API

One of fields was removed recently during schema updates, and should be
removed from documentation too.
Also add relevant bits about KCIDB_REST, including migration procedures.

Signed-off-by: Denys Fedoryshchenko <denys.f@collabora.com>
